src/static/js/webmentions.js

```diff
@@ -0,0 +1,211 @@
+    reactionA.setAttribute("title", reaction["author"]["name"]);
+    reactionA.setAttribute("aria-label", reaction["author"]["name"]);
+
+    const reactionIMG = document.createElement("img");
+    reactionIMG.setAttribute("class", "webmention-author-avatar");
+    reactionIMG.setAttribute("src", reaction["author"]["photo"]);
+    reactionIMG.setAttribute("alt", reaction["author"]["name"]);
+    reactionIMG.setAttribute("loading", "lazy");
+    reactionIMG.setAttribute("width", "60");
+    reactionIMG.setAttribute("height", "60");
+
+    reactionA.appendChild(reactionIMG);
+
+    // Replies and mentions have some extra HTML
+    const reactionDIVContent = document.createElement("div");
+    const reactionDIVMeta = document.createElement("div");
+    if (reactionType === "replies" || reactionType === "mentions") {
+      const reactionSTRONG = document.createElement("strong");
+      reactionSTRONG.setAttribute("class", "webmention-author-name");
+      reactionSTRONG.textContent = reaction["author"]["name"];
+      reactionA.appendChild(reactionSTRONG);
+
+      reactionDIVContent.setAttribute("class", "webmention-content");
+      reactionDIVContent.textContent = reaction["content"]["text"];
+
+      reactionDIVMeta.setAttribute("class", "webmention-meta");
+
+      const reactionTIME = document.createElement("time");
+      reactionTIME.setAttribute("class", "webmention-pub-date");
+      reactionTIME.setAttribute("datetime", reaction["published"]);
+      reactionTIME.textContent = formatDate(reaction["published"]);
+
+      const reactionSPAN = document.createElement("span");
+      reactionSPAN.setAttribute("class", "webmention-divider");
+      reactionSPAN.setAttribute("aria-hidden", "true");
+      reactionSPAN.textContent = " ⋅ ";
+
+      const reactionASource = document.createElement("a");
+      reactionASource.setAttribute("class", "webmention-source");
+      reactionASource.setAttribute("href", reaction["url"]);
+      reactionASource.textContent = document.querySelector(".reactions").getAttribute("data-source");
+
+      reactionDIVMeta.appendChild(reactionTIME);
+      reactionDIVMeta.appendChild(reactionSPAN);
+      reactionDIVMeta.appendChild(reactionASource);
+    }
+
+    reactionLI.appendChild(reactionA);
+    if (reactionType === "replies" || reactionType === "mentions") {
+      reactionLI.appendChild(reactionDIVContent);
+      reactionLI.appendChild(reactionDIVMeta);
+    }
+    webmentionReactionsList.appendChild(reactionLI);
+  });
+  document.querySelector(`#${reactionType}-panel`).appendChild(webmentionReactionsList);
+}
+
+// Parses and renders mentions into likes, reposts, replies and mentions
+function renderWebmentions(webmentions) {
+  if (!webmentions.length) {
+    return;
+  }
+
+  renderReactions(webmentions, "likes");
+  renderReactions(webmentions, "reposts");
+  renderReactions(webmentions, "replies");
+  renderReactions(webmentions, "mentions");
+
+}
```

Comment:
Is there much value in this function? Especially since you need have the `reactionMap` hardcoded above anyway so we know there are 4 types? Would it be better to merge renderWebmentions and `renderReactions into one function?

Reply:
That makes sense to me. I will do that.

Reply:
Done